### DISCUSSION TITLE
MQTT broker IP change (Home Assistant)

### DISCUSSION THREAD

**Discussion by helderem** [arendst/Tasmota#15548] [Q&A] [answered]
My tasmota devices are setup with MQTT using a hostname ("homeassistant"). The IP address of my MQTT broker has changed because of a router change. Tasmota devices are still connecting to the old IP address, only when I change the value to "homeassistant.local" the change is being picked up by the device.

I've also tried to change the value to "homeAssistant" but that is not picked up as a change, also changing the value to an IP address and then back to the old "homeassistant" value doesn't solve the issue.

Broken state:
![image](https://user-images.githubusercontent.com/16709595/166929221-0b63485c-a51b-492b-b727-8ba83d5c4bc4.png)

Fixed:
![image](https://user-images.githubusercontent.com/16709595/166929256-ee880b1d-662c-4a06-b868-a18e9a9d5dd5.png)

Is this an caching issue in tasmota or is there probably an issue with the DNS of my internal network?

**Comment by barbudor** [comment, marked as answer]
I would bet for something related to your router DNS configuration
Your previous router was probably not adding a local domain to the names while the new one does add ".local"
It may be configurable on your router.

**Reply by sfromis** [reply]
If you want to explore what Tasmota may (or not) have cached, just restart a device, to see if it then can connect. I'd not expect this to be a success.

**Comment by joba-1** [comment]
It would help to see which DNS server your windows machine (ipconfig) and your tasmota device (status 5) actually use.

Also, your windows pc could be configured to add a list of domains to the bare hostname to get answer from the dns server.

**Comment by Jason2866** [comment]
Using *.local is always a source for issues. Fire up your favorite search engine and do some research about.
